tests/fixtures/appFixture.ts

```diff
@@ -19,32 +20,56 @@ export const test = base.extend<{
     await use(app);
   },
 
-  loggedInApp: async ({ browser }, use) => {
-    console.log('📁 Generating new auth file at:', authFile);
-
-    const context = await browser.newContext();
-    const page = await context.newPage();
-    const app = new App(page);
-    await app.loginPage.gotoLoginPage();
-    await app.loginPage.performLogin(USER_EMAIL, USER_PASSWORD);
+  loggedInApp: async ({ browser, request }, use) => {
+    // 1. Login to get the token
+    const loginResponse = await request.post('https://api.practicesoftwaretesting.com/users/login', {
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      data: {
+        email: USER_EMAIL,
+        password: USER_PASSWORD,
+      },
+    });
 
-    await page.waitForURL('**/account');
+    if (loginResponse.status() !== 200) {
+      const errorBody = await loginResponse.text();
+      console.error(`❌ Login failed (${loginResponse.status()}): ${errorBody}`);
+      throw new Error('Login failed.');
+    }
 
-    const cookies = await context.cookies();
-    console.log('🍪 Cookies after login:', cookies.map(c => c.name));
+    let loginData: LoginResponse;
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      loginData = await loginResponse.json();
+    } catch (e) {
+      const raw = await loginResponse.text();
+      console.error('❌ Failed to parse login response:', e, '\nRaw:', raw);
+      throw new Error('Invalid JSON in login response.');
+    }
 
-    if (!fs.existsSync(authDir)) {
-      fs.mkdirSync(authDir, { recursive: true });
+    const token = loginData.token;
+    if (!token) {
+      throw new Error('❌ No token found in login response.');
     }
 
-    await context.storageState({ path: authFile });
-    console.log('💾 New auth file saved');
+    // 2. Open browser and inject token into localStorage
+    const context = await browser.newContext();
+    const page = await context.newPage();
+
+    await page.goto('https://api.practicesoftwaretesting.com/users/login'); 
```

Comment:
this URL is more suitable for request, but here we try to reach it via the browser.
```suggestion
    await page.goto('/users/login'); 
```